Tests/BotTests/MergeService/MergeServiceTests.swift

```diff
@@ -970,35 +1003,69 @@ class MergeServiceTests: XCTestCase {
+        case mergeServiceDestroyed(forBranch: String)
+
+        // Default values in enums were only introduced in Swift 5.1 so until we migrate…
+        static var mergeServiceCreated: DispatchServiceState { return .mergeServiceCreated(forBranch: MergeServiceFixture.defaultTargetBranch) }
+        static var mergeServiceDestroyed: DispatchServiceState { return .mergeServiceDestroyed(forBranch: MergeServiceFixture.defaultTargetBranch) }
+        static func mergeServiceState(_ state: MergeService.State) -> DispatchServiceState { return .mergeServiceState(branch: MergeServiceFixture.defaultTargetBranch, state: state) }
+    }
+
     private func perform(
         requiresAllStatusChecks: Bool = false,
         stubs: [MockGitHubAPI.Stubs],
         when: (MockGitHubEventsService, TestScheduler) -> Void,
-        assert: ([MergeService.State]) -> Void
+        assert: @escaping ([DispatchServiceState]) -> Void
     ) {
 
         let scheduler = TestScheduler()
         let gitHubAPI = MockGitHubAPI(stubs: stubs)
         let gitHubEvents = MockGitHubEventsService()
 
-        let service = MergeService(
+        let (states, statesObserver) = Signal<DispatchServiceState, NoError>.pipe()
+
+        let (lifetimeSignal, lifetimeObserver) = Signal<DispatchService.MergeServiceLifetime, NoError>.pipe()
+
+        self.dispatchService = DispatchService(
             integrationLabel: LabelFixture.integrationLabel,
             topPriorityLabels: LabelFixture.topPriorityLabels,
             requiresAllStatusChecks: requiresAllStatusChecks,
             statusChecksTimeout: MergeServiceFixture.defaultStatusChecksTimeout,
             logger: MockLogger(),
             gitHubAPI: gitHubAPI,
             gitHubEvents: gitHubEvents,
-            scheduler: scheduler
+            scheduler: scheduler,
+            mergeServiceLifetimeObserver: lifetimeObserver
         )
 
-        var states: [MergeService.State] = []
-
-        service.state.producer.observe(on: scheduler).startWithValues { states.append($0) }
+        lifetimeSignal
+            .observe(on: scheduler)
+            .observeValues { lifetime in
+                switch lifetime {
+                case .created(let service):
+                    statesObserver.send(value: .mergeServiceCreated(forBranch: service.targetBranch))
+                    service.state.producer
+                        .observe(on: scheduler)
+                        .startWithValues {
+                            statesObserver.send(value: .mergeServiceState(branch: service.targetBranch, state: $0))
+                        }
+                case .destroyed(let service):
+                    statesObserver.send(value: .mergeServiceDestroyed(forBranch: service.targetBranch))
+                }
+        }
 
         when(gitHubEvents, scheduler)
-        assert(states)
+
+        statesObserver.sendCompleted()
+        states.collect().observeValues(assert)
```

Comment:
Note: I'm not a fan of this approach but it seams that other approaches I've tried led to race conditions and thus an inconsistent order in the states that got collected.

For example I tried using an approach similar to `TestObserver` by having `var states: [DispatchServiceState] = []` then using `statesObserver.observeValues { states.append($0) }` and finally assert on `states` values – like [in this commit here](b8d5d847bf5ca3e8e91db52b69959bef8dd84438), but it led to the test failing, claiming that it got an order ending with `[…, runningStatusChecks, .idle, .ready]` – which is not possible because `reduceRunningStatusChecks` can't go to `.idle`, only to `.ready` (and only transition possible to `.idle` is from `.starting` or `.ready`), so clearly the issue was with the way of collecting state values…

Not sure how it can be that `.ready` and `.idle` could be swapped like that: it would probably mean that they're emitted one right after the other, during the same "heartbeat" of the Scheduler, but given that `.idle` state is supposed to only transition from `ready` using RAF after a `.noMorePullRequests` event, they should be emitted one after the other anyway, right?